apps/s_server.c

```diff
@@ -1500,7 +1506,16 @@ int s_server_main(int argc, char *argv[])
         case OPT_KEYLOG_FILE:
             keylog_file = opt_arg();
             break;
-
+        case OPT_MAX_EARLY:
+            max_early_data = atoi(opt_arg());
+            if (max_early_data < 0) {
+                BIO_printf(bio_err, "Invalid value for max_early_data\n");
+                goto end;
+            }
+            break;
+        case OPT_EARLY_DATA:
```

Comment:
No - max_early_data has an effect whether or not we accept or reject the early data. Either way it tells us how much data the client is allowed to send. If we reject then we're going to do trial decryption until we find a record that we can read and skip over the rest. If we've done trial decryption for max_early_data bytes and still haven't found a record we can read then we abort the handshake. If you don't also supply the "-early_data" flag then we are always going to reject the early data.